diff --git a/pynag.spec b/pynag.spec
index e5f5d88..e1834af 100644
--- a/pynag.spec
+++ b/pynag.spec
@@ -3,17 +3,6 @@
 %{!?python_version: %global python_version %(%{__python} -c "from distutils.sysconfig import get_python_version; print get_python_version()")}
 %endif
 
-# RHEL6 and newer has unittest2
-# All other distributions assume that we have access to unittest2
-%define unittest2 0
-%if 0%{?rhel} 
-%if 0%{?rhel} >= 6
-%define unittest2 1
-%endif
-%else
-%define unittest2 1
-%endif
-
 
 Summary: Python modules and utilities for Nagios plugins and configuration
 Name: pynag
@@ -27,9 +16,6 @@ BuildRequires: python-setuptools
 BuildRoot: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
 Url: http://pynag.org/
 BuildArch: noarch
-%if 0%{?unittest2}
-BuildRequires: python-unittest2
-%endif
 
 %description
 Python modules and utilities for pragmatically handling Nagios configuration
@@ -51,10 +37,6 @@ are scripts which list services, do network discovery among other tasks.
 %build
 %{__python} setup.py build
 
-%if 0%{?unittest2}
-%{__python} setup.py test
-%endif
-
 %install
 test "x$RPM_BUILD_ROOT" != "x" && rm -rf $RPM_BUILD_ROOT
 %{__python} setup.py install --prefix=/usr --root=$RPM_BUILD_ROOT
